src/helper/args/mod.rs

```diff
@@ -98,19 +96,30 @@ pub fn get_args_help<Output: Write>(
         if args.is_empty() {
             return Ok(());
         }
-        for arg_variants in [
-            (config.check_version).then(|| &args[0]),
-            (config.check_help && args.len() >= 2).then(|| &args[1]),
-        ]
+        for arg_variants in if args[0].is_empty() {
+            [(args.len() > 1).then(|| &args[1..]), None]
+        } else {
+            [
+                (config.check_version).then(|| &args[..1]),
+                (config.check_help && args.len() >= 2).then(|| &args[1..]),
+            ]
+        }
```

Comment:
The `test_check_help_args` test fail because this statement returns `[None, None]` which means no argument is checked.